community/louvain_common.go

```diff
@@ -35,11 +36,194 @@ func Q(g graph.Graph, communities [][]graph.Node, resolution float64) float64 {
 	}
 }
 
+// ReducedGraph is a modularised graph.
+type ReducedGraph interface {
+	graph.Graph
+
+	// Communities returns the community memberships
+	// of the nodes in the graph used to generate
+	// the reduced graph.
+	Communities() [][]graph.Node
+
+	// Structure returns the community structure of
+	// the current level of the module clustering.
+	// The first index of the returned value
```

Comment:
I've been reading this comment repeatedly for a couple of days and still I'm not sure what Structure returns. Isn't there a clearer formulation?